### DISCUSSION TITLE
Stuck in collection.load()

### DISCUSSION THREAD

**Discussion by HoangSang1510** [milvus-io/milvus#33059] [Q&A and General discussion] [answered]
Hello, i am using milvus 2.3.7, when i call collection.load() , it keep running, when i cancel command, it said that:
  File "/home/ims/.pyenv/versions/ims_id/lib/python3.10/site-packages/pymilvus/client/grpc_handler.py", line 1074, in wait_for_loading_collection
    time.sleep(Config.WaitTimeDurationWhenLoad)
KeyboardInterrupt

How can i fix it, thanks you

**Comment by yhmo** [comment, marked as answer]
collection.load() will call get_loading_progress() to check the loading state of the collection, in the interval for each 1 second.
If collection.load() is stuck for long time, two possible reasons:
1. the collection is huge, loading process needs to read all the data files from S3/MinIO, disk read performance is poor, or the bandwidth between Milvus and remote S3 is poor.
2. something is wrong with the Milvus server, get the full log of the Milvus server to investigate.
